### PR TITLE
Implement Receiver feature

### LINKED ISSUE #5: Implementation of Interface IERC3525Receiver
**Description**
---
Implement the functions of the Interface `IERC3525Receiver` of the [EIP-3525](https://eips.ethereum.org/EIPS/eip-3525). 

This interface is implemented in an external smart contract that wants to be informed by EIP-3525 contracts when receiving values from ANY addresses or EIP-3525 tokens.

The list of functions are :
- `onERC3525Received`

**Acceptance Criteria**
---
- [x] Implement the contents of each functions and events 
- [x] Implement unit tests for each functions
- [x] Implement integration tests for each functions
- [ ] If needed, write the documentation associated with the functions and events exposed

### PR DESCRIPTION
**Pull request by tekkac**
Closes #5

### CODE REVIEW

File: tests/unit/library.cairo [hunk 1]
```diff
@@ -77,6 +78,8 @@ namespace it {
 
         %{
             stop_prank = start_prank(ids.caller) 
+            mock_call(ids.to, "onERC3525Received", [ids.IERC3525_RECEIVER_ID])
+            mock_call(ids.to, "supportsInterface", [1])
```
Comment (cloudvenger) [inline tests/unit/library.cairo]: to be consistent with the other test, do we want to change `[1]` to `[ids.TRUE]`  ?

File: tests/unit/library.cairo [hunk 2]
```diff
@@ -114,7 +117,9 @@ namespace it {
         let (local old_slot) = ERC3525.slot_of(from_token_id);
 
         %{
-            stop_prank = start_prank(ids.caller) 
+            stop_prank = start_prank(ids.caller)
+            mock_call(ids.to, "onERC3525Received", [ids.IERC3525_RECEIVER_ID])
+            mock_call(ids.to, "supportsInterface", [1])
```
Comment (cloudvenger) [inline tests/unit/library.cairo]: to be consistent with the other test, do we want to change `[1]` to `[ids.TRUE]`  ?

File: tests/unit/library.cairo [hunk 3]
```diff
@@ -154,6 +159,8 @@ namespace it {
 
         %{
             stop_prank = start_prank(ids.caller) 
+            mock_call(ids.old_to_owner, "onERC3525Received", [ids.IERC3525_RECEIVER_ID])
+            mock_call(ids.old_to_owner, "supportsInterface", [1])
```
Comment (cloudvenger) [inline tests/unit/library.cairo]: to be consistent with the other test, do we want to change `[1]` to `[ids.TRUE]`  ?

File: tests/unit/library.cairo [hunk 4]
```diff
@@ -191,6 +198,8 @@ namespace it {
         let (local old_slot) = ERC3525.slot_of(from_token_id);
         %{
             stop_prank = start_prank(ids.caller) 
+            mock_call(ids.old_owner, "onERC3525Received", [ids.IERC3525_RECEIVER_ID])
+            mock_call(ids.old_owner, "supportsInterface", [1])
```
Comment (cloudvenger) [inline tests/unit/library.cairo]: to be consistent with the other test, do we want to change `[1]` to `[ids.TRUE]`  ?

**Review by cloudvenger — DISMISSED** [review]
looks good to me :) 

Just some minor comment

File: tests/unit/library.cairo (hunk 1 above)
Comment (tekkac) [reply tests/unit/library.cairo]: good catch

**Review by cloudvenger — APPROVED** [review]

**Pull request merged**